### PR TITLE
ch4/ofi: am related clean up

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description
This PR cleans up the ofi am related routines, mainly the CQ buffering routines and am repost routines. CQ buffering is needed during am send operation when we need poke progress but should not process the cq entries, or we may end up with deep nesting progress. The am repost is required to ensure am packet ordering is preserved at target callbacks. This is because the FI_MULT_RECV buffer doesn't preserve order upon generating CQ entries. The clean-up mainly focuses on grouping related routines together so they can be more easily maintained.

It prepares for enabling multiple vci in the am path.

[skip warnings]

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
test:mpich/ch4/ofi

File: src/mpid/ch4/netmod/ofi/ofi_pre.h [hunk 1]
```diff
@@ -150,8 +150,9 @@ typedef struct MPIDI_OFI_deferred_am_isend_req {
     MPI_Datatype datatype;
     MPIR_Request *sreq;
     bool need_packing;
-
     MPI_Aint data_sz;
+    int vni_src;
+    int vni_dst;
```
Comment (raffenet) [inline src/mpid/ch4/netmod/ofi/ofi_pre.h:155]: unused vars? or maybe leaked from another commit?

**Review by raffenet — COMMENTED** [review]
Just a couple of comments.

File: src/mpid/ch4/netmod/ofi/ofi_am_impl.h [hunk 2]
```diff
@@ -14,7 +14,18 @@
...
 
-/* Acquire a sequence number to send, and record the next number */
+/* active message ordering --
+ * FI_MULTI_RECV doesn't guarantee the ordering of messages. Thus we have to
+ * set and check the seqno and potentially enqueue the unordered messages.
+ * This is necessary because CH4 message queue relies on transport observing the * message order.
+ *
+ * Each time we send an active message, we increment and embed a send_seqno in
+ * the msg_hdr. At the receiver end we maintain a recv_seqno and match against
+ * the send_seqno. If they are out-of-order, we post point the message by
```
[the remark this answers is not in the document]
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_am_impl.h]: Good catch! Fixed.

File: src/mpid/ch4/netmod/ofi/ofi_pre.h (hunk 1 above)
Comment (hzhou) [reply src/mpid/ch4/netmod/ofi/ofi_pre.h]: Yes, it was leaked in. Now removed.

**Review by raffenet — APPROVED** [review]

**Pull request merged**